src/libtiled/maprenderer.cpp

```diff
@@ -335,7 +350,17 @@ void CellRenderer::render(const Cell &cell, const QPointF &pos, const QSizeF &si
     const QRectF source(0, 0, fragment.width, fragment.height);
 
     mPainter->setTransform(transform);
-    mPainter->drawPixmap(target, image, source);
+    QPixmap tintedimg = image;
+    if(layer){
+        QColor tintColor = layer->tintColor();
+        if(tintColor != QColor(255, 255, 255, 255)){
+            tintedimg = QPixmap::fromImage(tinted(QImage(image.toImage())
+                                                    ,layer->tintColor()
```

Comment:
Either use the local `tileColor` here or use `layer->tintColor` also in the condition. I would probably remove the local `tintColor`.

Reply:
done